src/rviz/InteractiveMarkerViewer.cpp

```diff
@@ -49,6 +49,39 @@ FrameMarkerPtr InteractiveMarkerViewer::addFrame(
   return marker;
 }
 
+
+
+void InteractiveMarkerViewer::visualizeTSR(const aikido::constraint::TSR& _tsr,
+  int nSamples)
+{
+  using dart::dynamics::Frame;
+  using dart::dynamics::SimpleFrame;
+  using aikido::constraint::TSR;
+
+  auto sampler = _tsr.createSampleGenerator();
+  auto state = _tsr.getSE3()->createState();
+
+  for(int i = 0; i < nSamples && sampler->canSample(); ++i)
+  {
+    sampler->sample(state);
+
+    std::stringstream sstm;
+    sstm << "tsr" << i;
```

Comment:
Suggestion: Make the base name a `std::string` parameter. I suggest making it default to something that includes the `&tsr` (or, even better, the memory address of the return value) to avoid name collisions.